core/src/test/java/io/questdb/cairo/WriteAndReadSyncTest.java

```diff
@@ -0,0 +1,114 @@
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ ******************************************************************************/
+
+package io.questdb.cairo;
+
+import io.questdb.log.Log;
+import io.questdb.log.LogFactory;
+import io.questdb.std.Files;
+import io.questdb.std.FilesFacade;
+import io.questdb.std.FilesFacadeImpl;
+import io.questdb.std.Unsafe;
+import io.questdb.std.str.Path;
+import org.junit.Assert;
+import org.junit.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.Random;
+import java.util.concurrent.BrokenBarrierException;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.CyclicBarrier;
+import java.util.concurrent.atomic.AtomicInteger;
+
+public class WriteAndReadSyncTest extends AbstractCairoTest {
+
+    private final static Log LOG = LogFactory.getLog(WriteAndReadSyncTest.class);
+
+    @Test
+    public void testVanilla() throws IOException, BrokenBarrierException, InterruptedException {
+        final long longsPerPage = Files.PAGE_SIZE / 8;
+        for(int loop = 0; loop < 10; loop++) {
+            Random rnd = new Random();
+            // increments randomly by [0..512] up to three pages
+            for (long longCountIncr = longsPerPage, limit = longsPerPage * 3; longCountIncr < limit; longCountIncr += rnd.nextDouble() * 512) {
+                long longCount = longCountIncr;
+                final CountDownLatch readLatch = new CountDownLatch(1);
+                final File file = temp.newFile();
+                try (Path path = new Path()) {
+                    path.of(file.getAbsolutePath()).$();
+                    FilesFacade ff = FilesFacadeImpl.INSTANCE;
+
+
+                    // barrier to make sure both threads kick in at the same time;
+                    final CyclicBarrier barrier = new CyclicBarrier(2);
+                    final AtomicInteger errorCount = new AtomicInteger();
+                    long fd1 = TableUtils.openRW(ff, path, LOG);
+                    long size = longCount * 8 / Files.PAGE_SIZE + 1;
+
+                    // have this thread write another page
+                    Thread th = new Thread(() -> {
+                        try {
+                            barrier.await();
+                            // over allocate
+                            long mem = TableUtils.mapRW(ff, fd1, (size) * Files.PAGE_SIZE);
+                            for (int i = 0; i < longCount; i++) {
+                                Unsafe.getUnsafe().putLong(mem + i * 8L, i);
+                            }
+                            readLatch.countDown();
+                            ff.munmap(mem, (size) * Files.PAGE_SIZE);
```

Comment:
https://linux.die.net/man/2/munmap
```
The munmap() system call deletes the mappings for the specified address range, and causes further references to addresses within the range to generate invalid memory references. The region is also automatically unmapped when the process is terminated. On the other hand, closing the file descriptor does not unmap the region.
```
so that would mean that mem is now pointing to unmapped memory. The next call is to truncate, will the content pointed to by mem be still there?

Reply:
Content will be in the file, as if you wrote to the file.

We're testing this  https://man7.org/linux/man-pages/man2/mmap.2.html

>   The effect of changing the size of the underlying file of a
>        mapping on the pages that correspond to added or removed regions
>        of the file is unspecified.